hazelcast-hibernate53/src/main/java/com/hazelcast/hibernate/local/FreeHeapBasedCacheEvictor.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.hazelcast.hibernate.local;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Policy;
+import com.hazelcast.internal.util.MemoryInfoAccessor;
+import com.hazelcast.internal.util.RuntimeMemoryInfoAccessor;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+class FreeHeapBasedCacheEvictor implements AutoCloseable {
+    private static final int TERMINATE_TIMEOUT_SECONDS = 5;
+    private static final Duration DEFAULT_EVICTION_DELAY = Duration.ofSeconds(1);
+    private static final int EVICTION_BATCH_SIZE = 15;
+    private final ILogger log = Logger.getLogger(getClass());
+
+    private final ScheduledExecutorService executorService;
+    private final MemoryInfoAccessor memoryInfoAccessor;
+    private final Duration evictionDelay;
+    private final AtomicBoolean started = new AtomicBoolean();
+
+    FreeHeapBasedCacheEvictor() {
+        this(Executors.newSingleThreadScheduledExecutor(defaultThreadFactory()), new RuntimeMemoryInfoAccessor(),
```

Comment:
I would prefer static import for `newSingleThreadScheduledExecutor` across the whole PR.